src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts

```diff
@@ -1718,4 +1768,93 @@ export class NetworkAreaDiagramViewer {
         const zoom = this.svgDraw?.zoom() ?? 1;
         this.svgDraw?.zoom(0.9 * zoom);
     }
+
+    public saveSvg() {
+        this.addStyle();
+        const blobData = [this.getSvg() ?? ''];
+        const blob = new Blob(blobData, { type: 'image/svg+xml' });
+        this.downloadFile(blob, 'nad.svg');
+        this.removeStyle();
+    }
+
+    private downloadFile(blob: Blob, filename: string) {
+        const a = document.createElement('a');
+        a.download = filename;
+        a.href = URL.createObjectURL(blob);
+        a.click();
+        a.remove();
+    }
+
+    private addStyle() {
+        // add style, if not present
+        if (this.svgParameters.getCssLocation() == CssLocationEnum.EXTERNAL_NO_IMPORT) {
+            const styleElement = DiagramUtils.getStyle(document.styleSheets, this.svgDraw?.node);
+            const gElement = this.svgDraw?.node.querySelector('g');
+            gElement?.before(styleElement);
+        }
+    }
+
+    private removeStyle() {
+        // remove style, if added
+        if (this.svgParameters.getCssLocation() == CssLocationEnum.EXTERNAL_NO_IMPORT) {
+            const styleElement: HTMLElement | null = this.svgDiv.querySelector('style');
+            styleElement?.remove();
+        }
+    }
+
+    public savePng(backgroundColor?: string) {
+        this.copyPng(true, backgroundColor);
+    }
+
+    public screenshot(backgroundColor?: string) {
+        this.copyPng(false, backgroundColor);
+    }
+
+    private copyPng(copyToFile: boolean, backgroundColor?: string) {
+        this.addStyle();
+        this.addBackgroundColor(backgroundColor);
+        const svgXml = DiagramUtils.getSvgXml(this.getSvg());
```

Comment:
Could you make a call to DiagramUtils.getViewBox before to get the zoom-to-fit svg to override the viewbox for the downloaded svg only? I think it makes more sense to have the svg not cropped to the currently zone (it's always anyway containing the full graph)